modules/server_new/src/main/scala/observe/server/engine/Event.scala

```diff
@@ -0,0 +1,126 @@
+
+package observe.engine
+
+import cats.effect.Sync
+import cats.syntax.all.*
+import fs2.Stream
+import lucuma.core.enums.Breakpoint
+import lucuma.core.model.User
+import lucuma.core.model.sequence.Step
+import observe.model.ClientId
+import observe.model.Observation
+import observe.server.EngineState
+import observe.server.SeqEvent
+
+import java.time.Instant
+
+import SystemEvent.*
+import UserEvent.*
+
+/**
+ * Anything that can go through the Event Queue.
+ */
+sealed trait Event[F[_]] extends Product with Serializable
+
+object Event {
+  case class EventUser[F[_]](ue: UserEvent[F])  extends Event[F]
+  case class EventSystem[F[_]](se: SystemEvent) extends Event[F]
+
+  def start[F[_]](id: Observation.Id, user: User, clientId: ClientId): Event[F] =
+    EventUser[F](Start[F](id, user.some, clientId))
+  def pause[F[_]](id: Observation.Id, user: User): Event[F]                     =
+    EventUser[F](Pause(id, user.some))
+  def cancelPause[F[_]](id: Observation.Id, user: User): Event[F]               =
+    EventUser[F](CancelPause(id, user.some))
+  def breakpoints[F[_]](
+    id:    Observation.Id,
+    user:  User,
+    steps: List[Step.Id],
+    v:     Breakpoint
+  ): Event[F] = EventUser[F](Breakpoints(id, user.some, steps, v))
+  def poll[F[_]](clientId: ClientId): Event[F]                                  =
+    EventUser[F](Poll(clientId))
+  def getState[F[_]](f: EngineState[F] => Stream[F, Event[F]]): Event[F]        =
+    EventUser[F](GetState(f))
+  def modifyState[F[_]](f: EngineHandle[F, SeqEvent]): Event[F]                 =
+    EventUser[F](ModifyState(f))
+  def actionStop[F[_]](
+    id: Observation.Id,
+    f:  EngineState[F] => Stream[F, Event[F]]
+  ): Event[F] = EventUser[F](ActionStop(id, f))
+  def actionResume[F[_]](
+    id: Observation.Id,
+    i:  Int,
+    c:  Stream[F, Result]
+  ): Event[F] =
+    EventUser[F](ActionResume(id, i, c))
+  def logDebugMsg[F[_]](msg: String, ts: Instant): Event[F]                     =
+    EventUser[F](LogDebug(msg, ts))
+  def logDebugMsgF[F[_]: Sync](msg: String): F[Event[F]]                        =
+    Sync[F].delay(Instant.now).map(t => EventUser[F](LogDebug(msg, t)))
+  def logInfoMsg[F[_]](msg: String, ts: Instant): Event[F]                      =
+    EventUser[F](LogInfo(msg, ts))
+  def logWarningMsg[F[_]](msg: String, ts: Instant): Event[F]                   =
+    EventUser[F](LogWarning(msg, ts))
+  def logErrorMsg[F[_]](msg: String, ts: Instant): Event[F]                     =
+    EventUser[F](LogError(msg, ts))
+  def logErrorMsgF[F[_]: Sync](msg: String): F[Event[F]]                        =
+    Sync[F].delay(Instant.now).map(t => EventUser[F](LogError(msg, t)))
+
+  def pure[F[_]: Sync](v: SeqEvent): Event[F] = EventUser[F](Pure(v))
+
+  def failed[F[_]](id: Observation.Id, i: Int, e: Result.Error): Event[F]  =
+    EventSystem[F](Failed(id, i, e))
+  def completed[F[_], R <: Result.RetVal](
+    id:     Observation.Id,
```

Comment:
Could we either make all parameters have a meaningful name or none?

Reply:
The goal is to have meaningful names everywhere, but that may take time. I've now renamed these.